DXMainClient/DXGUI/Generic/GameInProgressWindow.cs

```diff
@@ -131,41 +152,223 @@ private void HandleGameProcessExited()
                 GameClass.SetGraphicsMode(WindowManager);
             }
 
-            try
+            DateTime dtn = DateTime.Now;
+
+#if ARES
+            Task.Factory.StartNew(ProcessScreenshots);
+
+            // TODO: Ares debug log handling should be addressed in Ares DLL itself.
+            // For now the following are handled here:
+            // 1. Make a copy of syringe.log in debug snapshot directory on both crash and desync.
+            // 2. Move SYNCX.txt from game directory to debug snapshot directory on desync.
+            // 3. Make a debug snapshot directory & copy debug.log to it on desync even if full crash dump wasn't created.
+            // 4. Handle the empty snapshot directories created on a crash if debug logging was disabled.
+
+            string snapshotDirectory = GetNewestDebugSnapshotDirectory();
+            bool snapshotCreated = snapshotDirectory != null;
+
+            snapshotDirectory = snapshotDirectory ?? ProgramConstants.GamePath + "debug/snapshot-" +
+                dtn.ToString("yyyyMMdd-HHmmss");
+
+            bool debugLogModified = false;
+            string debugLogPath = ProgramConstants.GamePath + "debug/debug.log";
+            DateTime lastWriteTime = new DateTime();
+
+            if (File.Exists(debugLogPath))
+                lastWriteTime = File.GetLastWriteTimeUtc(debugLogPath);
+
+            if (!lastWriteTime.Equals(debugLogLastWriteTime))
             {
-                if (!Directory.Exists(ProgramConstants.GamePath + "Client/ErrorLogs"))
-                    Directory.CreateDirectory(ProgramConstants.GamePath + "Client/ErrorLogs");
+                debugLogModified = true;
+                debugLogLastWriteTime = lastWriteTime;
+            }
 
-                DateTime dtn = DateTime.Now;
+            if (CopySyncErrorLogs(snapshotDirectory, null) || snapshotCreated)
+            {
+                if (File.Exists(debugLogPath) && !File.Exists(snapshotDirectory + "/debug.log") && debugLogModified)
+                    File.Copy(debugLogPath, snapshotDirectory + "/debug.log");
 
-                if (File.Exists(ProgramConstants.GamePath + "EXCEPT.TXT"))
+                CopyErrorLog(snapshotDirectory, "syringe.log", null);
+            }
+#else
+            if (deletingLogFilesFailed)
+                return;
+
+            CopyErrorLog(ProgramConstants.GamePath + "Client/ErrorLogs", "EXCEPT.TXT", dtn);
+            CopySyncErrorLogs(ProgramConstants.GamePath + "Client/ErrorLogs", dtn);
```

Comment:
I wonder if we should copy the error logs in what seems to be the client-only stuff folder?

Reply:
As a compromise and to not clutter the main game directory with multiple more subdirectories, they are now saved to separate subdirectories within the Client directory.